test/node/agent.js

```diff
@@ -28,23 +29,36 @@ function startServer (handler) {
 
       resolve({
         port: addressInfo && (typeof addressInfo === 'string' ? addressInfo : addressInfo.port),
-        close: () => server.close()
+        close: (...args) => server.close(...args)
       })
     })
   })
 }
 
-describe('agent', function () {
+/**
+ * This test was discovered as broken during https://github.com/ipfs/js-kubo-rpc-client/pull/83
+ * the test has many problems:
+ *
+ * 1. It's not deterministic
+ * 2. It's not actually validating that concurrent tests aren't allowed
+ * 3. ipfs.id() does not forward the agent property.
+ *
+ * I spent some time debugging this and was unable to resolve, as this was already an inherent problem with ipfs-http-client, i'm punting for now.
+ *
+ * @see https://github.com/ipfs/js-kubo-rpc-client/tree/investigateConcurrencyTest
+ */
+// eslint-disable-next-line mocha/no-skipped-tests
+describe.skip('agent', function () {
```

Comment:
:thinking: I think this is validating that only two unresolved requests are in-flight at the same time.

Perhaps to be more humane around browser limits of connections per domain?
Capping concurrency at two makes sense as a way to not overwhelm the global request limit on a web page, i guess.

Reply:
I just realized I need to add this to #56 .

I attempted to update this code to be more deterministic, but was unable to do so without significant changes, and it's pretty intermittent when it is passing. I see the value of this test, but I think it needs a complete rewrite. Is this something you're comfortable with leaving skipped and addressing later?

Reply:
also, when diving into the requests themselves, ALL of the tests in flight (in existing test, and when I tested) were sending connection requests to the server. They were ALL being kept-alive. I think I tested up to 10 requests. Ideally, if we set maxSockets to 1, and maxRequestsPerSocket to 1, then there should only ever be one request that fires an event on the server, blocking others, but that was not the case. They would all fire all the same events, and resolving any response for any of them would result in that request being completed.

For a limited set of requests, you would expect that only the two "in flight" requests could have their responses resolved at any given time, but instead, any will respond as soon as you resolve it. This means that limiting concurrent requests is not currently supported. open to learning if my expectations are wrong; this problem took three days from me.

Reply:
see https://github.com/ipfs/js-kubo-rpc-client/issues/98